scripts/release/change_sonarcloud_project_version.js

```diff
@@ -0,0 +1,36 @@
+const { readFileSync, writeFileSync } = require('node:fs')
+const path = require('node:path')
+
+// eslint-disable-next-line import/no-unresolved
+const { PropertiesEditor } =require("properties-file/editor")
+
+
+const SONAR_PROJECT_PROPERTIES_FILE = path.join(__dirname, "../../sonar-project.properties")
+
+const newVersion = process.argv[2]
+if (!newVersion) {
+  throw new Error("No version specified")
+}
+// Sanitise the version number.
+const versionRegex = /^[vV]\d+\.\d+\.\d+$/
+if (!versionRegex.test(newVersion)) {
+  throw new Error(`Version ${newVersion} does not match ${versionRegex}`)
+}
+
+// Update the version in memory.
+const properties = new PropertiesEditor(readFileSync(SONAR_PROJECT_PROPERTIES_FILE, "utf8"))
+properties.update("sonar.projectVersion", {
+  newValue: newVersion,
+})
+
+
+//console.log(properties.format())
+// Write the file back to disk.
+try {
+  writeFileSync(SONAR_PROJECT_PROPERTIES_FILE, properties.format())
```

Comment:
Nice .. some formatting! I can't seem to get a formatter for this file in vs code